.travis.yml

```diff
@@ -89,45 +114,70 @@ install:
       conda update -q conda
     fi
   - |
-    if [[ "$RUNMODE" != "docker" ]] ; then
+    if [[ "$RUNMODE" != "docker" ]] && [[ "$RUNMODE" != "singularity" ]]; then
       # Install conda packages
       ${TRAVIS_BUILD_DIR}/contrib/devenv/create.sh
       source ${TRAVIS_BUILD_DIR}/build/env/start
       ${TRAVIS_BUILD_DIR}/contrib/conda.sh
       ${TRAVIS_BUILD_DIR}/contrib/build-pybind11-in-conda.sh
       # Debugging information
       conda info -a
-    else
+    elif [[ "$RUNMODE" == "docker"  ]]; then
       $SCRIPT_RUN_PREFIX id
       $SCRIPT_RUN_PREFIX apt-get -qq update
       $SCRIPT_RUN_PREFIX apt-get -qqy install fakeroot debhelper locales \
         libreadline7 libssl1.0.0 libffi6 \
         liblapack3 liblapack-dev libhdf5-100 libhdf5-dev libnetcdf13 \
         libnetcdf-dev libscotch-6.0 libscotch-dev cython3 python3 \
         python3-numpy libpython3.6-dev python3-boto python3-paramiko graphviz
+    elif [[ "$RUNMODE" == "singularity"  ]]; then
+      pushd ${SHOME}
+      cp ${TRAVIS_BUILD_DIR}/sandbox/gas/tube/${SOLVCON_DRIVING} ${SHOME}
+      # trick to inject TRAVIS_BRANCH and TRAVIS_REPO_SLUG
+      echo '#!/bin/bash' > ${SHOME}/prepare-solvcon-dev.sh
+      echo "export TRAVIS_BRANCH=${TRAVIS_BRANCH}" >> ${SHOME}/prepare-solvcon-dev.sh
+      echo "export TRAVIS_REPO_SLUG=${TRAVIS_REPO_SLUG}" >> ${SHOME}/prepare-solvcon-dev.sh
+      cat ${TRAVIS_BUILD_DIR}/contrib/singularity/prepare-solvcon-dev.sh >> ${SHOME}/prepare-solvcon-dev.sh
+      cp ${TRAVIS_BUILD_DIR}/contrib/singularity/${SRECIPE} ${SHOME}
+      sudo singularity build ${SIMAGE} ${SRECIPE}
+      ls
+      popd
     fi
 
 script:
   # Debugging information
   - |
-    $SCRIPT_RUN_PREFIX which python3
-    $SCRIPT_RUN_PREFIX python3 --version
-    $SCRIPT_RUN_PREFIX which $CXX
-    $SCRIPT_RUN_PREFIX $CXX --version
-    $SCRIPT_RUN_PREFIX which $CC
-    $SCRIPT_RUN_PREFIX $CC --version
-  - $SCRIPT_RUN_PREFIX make SC_PURE_PYTHON=1 test_from_package
+    # We do everything in the singularity container including unit tests. The
```

Comment:
We don't need to change this.  Having the information printed isn't harmful.

Reply:
I'd prefer the debugging information be printed for all runs.

And If `python3` isn't in the path, there's a bug.  The executable should be always available in the path.